test/adtests.jl

```diff
@@ -1174,8 +1174,8 @@ using MLUtils
     optf = OptimizationBase.instantiate_function(
         optf, rand(3), AutoEnzyme(), iterate(data)[1], g = true, fg = true)
     G0 = zeros(3)
-    @test_broken optf.grad(G0, ones(3), (x, y))
-    stochgrads = []
+    @test_broken optf.grad(G0, ones(3), (x0, y0))
```

Comment:
@wsmoses since you might have the branch already checked out, can you try this out?

It gives 

```
  Constant memory is stored (or returned) to a differentiable variable.
  As a result, Enzyme cannot provably ensure correctness and throws this error.
  This might be due to the use of a constant variable as temporary storage for active memory (https://enzyme.mit.edu/julia/stable/faq/#Runtime-Activity).
  If Enzyme should be able to prove this use non-differentable, open an issue!
  To work around this issue, either:
   a) rewrite this variable to not be conditionally active (fastest, but requires a code change), or
   b) set the Enzyme mode to turn on runtime activity (e.g. autodiff(set_runtime_activity(Reverse), ...) ). This will maintain correctness, but may slightly reduce performance.
  Mismatched activity for:   store {} addrspace(10)* %unbox.unpack, {} addrspace(10)** %.fca.0.1.0.gep, align 8, !dbg !26, !noalias !56 const val:   %unbox.unpack = load {} addrspace(10)*, {} addrspace(10)* addrspace(11)* %unbox.elt, align 8, !dbg !26, !tbaa !23, !alias.scope !27, !noalias !30
   value=Unknown object of type Vector{Float64}
   llvalue=  %unbox.unpack = load {} addrspace(10)*, {} addrspace(10)* addrspace(11)* %unbox.elt, align 8, !dbg !26, !tbaa !23, !alias.scope !27, !noalias !30
  
  Stacktrace:
   [1] loss
     @ ~/OptimizationBase.jl/test/adtests.jl:1120
```

Reply:
It would be neat since this will allow training NNs with Enzyme + Optimization

Reply:
This test doesn't have an NN it tests the ability to pass additional arguments to the gradient call

Reply:
So here you should pass mode = set_runtime_activity(Reverse) inside the autoenzyme constructor